strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/io/ArtifactOutputStreamTest.java

```diff
@@ -49,49 +45,20 @@
         extends TestCaseWithMavenArtifactGenerationAndIndexing
 {
 
-    private static final String REPOSITORY_RELEASES = "aos-releases";
-
     @Inject
     private RepositoryPathResolver repositoryPathResolver;
 
-    private Set<MutableRepository> getRepositoriesToClean(TestInfo testInfo)
-    {
-        Set<MutableRepository> repositories = new LinkedHashSet<>();
-        repositories.add(createRepositoryMock(STORAGE0, getRepositoryName(testInfo), Maven2LayoutProvider.ALIAS));
-
-        return repositories;
-    }
-
-    private String getRepositoryName(TestInfo testInfo)
-    {
-        Optional<Method> method = testInfo.getTestMethod();
-        return REPOSITORY_RELEASES + method.get().getName();
-    }
-
-    @BeforeEach
-    public void initialize(TestInfo testInfo)
-            throws Exception
-    {
-        createRepository(STORAGE0, getRepositoryName(testInfo), false);
-
-        generateArtifact(getRepositoryBasedir(STORAGE0, getRepositoryName(testInfo)).getAbsolutePath(),
-                         "org.carlspring.foo:temp-file-test:1.2.3");
-    }
-
-    @AfterEach
-    public void removeRepositories(TestInfo testInfo)
-            throws IOException, JAXBException
-    {
-        removeRepositories(getRepositoriesToClean(testInfo));
-    }
 
     @Test
-    public void testCreateWithTemporaryLocation(TestInfo testInfo)
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class, ArtifactManagementTestExecutionListener.class })
+    public void testCreateWithTemporaryLocation(@TestRepository(repository = "aost=tcwtl-releases",
+                                                                layout = Maven2LayoutProvider.ALIAS)
+                                                Repository repository,
+                                                @MavenTestArtifact(id = "org.carlspring.foo:temp-file-test",
+                                                                   versions = { "1.2.3" })
+                                                Path path)
             throws IOException
     {
-        final Storage storage = getConfiguration().getStorage(STORAGE0);
-        final Repository repository = storage.getRepository(getRepositoryName(testInfo));
-
         final Artifact artifact = ArtifactUtils.getArtifactFromGAVTC("org.carlspring.foo:temp-file-test:1.2.3:jar");
         final ArtifactCoordinates coordinates = new MavenArtifactCoordinates(artifact);
```

Comment:
you can use `RepositoryFiles.readArtifactCoordinates(path.nozmalize())` here to get the `ArtifactCoordinates`.

Reply:
Do you think you could put together a brief article in the Developer's Guide on "How To Work With Our Filesystem And Paths".

Perhaps you could gather a few such common case usage examples and add a short description. This will help people get a better understanding of how to work with our API.

Reply:
ok

Reply:
Do you mean:
```
final ArtifactCoordinates coordinates = RepositoryFiles.readCoordinates(path.normalize());
```
...or what? It looks like this won't work. Am I doing it wrong, or do you have other suggestions?